### PR TITLE
Switch MCUmgr to use boot_set_next function for selecting next slot to boot

### PR DESCRIPTION
**Pull request by de-nordic**
The PR consists of commits:
 1) (To be removed) Update of MCUboot to bring in version providing boot_set_next;
 2) Providing `img_mgmt_active_slot(int image)` function, which allows to check what is active slot for given image and `img_mgmt_active_image() function which allows to get number of currently running image.
 3) Altering MCUmgr code to use `boot_set_next()` to select next slot to boot;

This changes have been tested with MCUboot in swap-move and DIrectXIP mode and they allow MCUmgr to correctly upload image and erase.
In case of DirectXIP does not work, but addressing full DirectXIP was not target of this PR.
Marked as DNM till the 1) gets merged into MCUboot Zephyr fork.

** Update 2023-05-09 **
Slot detection no longer relies on DT_CHOSEN and now uses slot offset. This allows custom flash map definitions to work with MCUmgr correctly.

### CODE REVIEW

**Comment by nordicjm** [comment]
Needs rebase

File: subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c [hunk 1]
```diff
@@ -267,6 +270,53 @@ img_mgmt_state_read(struct smp_streamer *ctxt)
...
+		}
+	}
+
+	/* Confirm disallowed if a test is pending. */
+	if (active && img_mgmt_state_any_pending()) {
+		return MGMT_ERR_EBADSTATE;
+	}
+
+	if (flash_area_open(area_id, &fa) != 0) {
+		return MGMT_ERR_EBADSTATE;
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c:295]: `Current state disallows command.` doesn't really seem apt here

File: subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c [hunk 2]
```diff
@@ -267,6 +270,53 @@ img_mgmt_state_read(struct smp_streamer *ctxt)
...
+		if (active) {
+			LOG_ERR("Failed to write pending flag for slot %d: %d", slot, rc);
+		} else {
+			LOG_ERR("Failed to write confirmed flag: %d", rc);
+		}
+		rc = MGMT_ERR_EUNKNOWN;
+	}
+	flash_area_close(fa);
+
+		/* Confirm event is only sent for active slot */
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c:310]: Wrong indent

File: subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c (hunk 1 above)
Comment (de-nordic) [reply subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c:295]: True. Fixing.

File: subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c [hunk 3]
```diff
@@ -267,6 +270,53 @@ img_mgmt_state_read(struct smp_streamer *ctxt)
...
+		return MGMT_ERR_EBADSTATE;
+	}
+
+	rc = boot_set_next(fa, active, confirm);
+	if (rc != 0) {
+		/* Failed to set next slot for boot as desired */
+		if (active) {
+			LOG_ERR("Failed to write pending flag for slot %d: %d", slot, rc);
+		} else {
+			LOG_ERR("Failed to write confirmed flag: %d", rc);
```
[the remark this answers is not in the document]
Comment (de-nordic) [reply subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c:304]: Yep. Fixing.

**Review by nordicjm — DISMISSED** [review]

**Comment by nordicjm** [comment]
Needs rebase

**Comment by de-nordic** [comment]
@carlescufi @MarekPieta @nordicjm Can you take a look?

**Review by nordicjm — APPROVED** [review]

**Pull request closed without merging**